src/routes/zome_call.rs

```diff
@@ -0,0 +1,24 @@
+use axum::extract::{Path, Query};
+use serde::Deserialize;
+
+#[derive(Debug, Deserialize)]
+#[allow(unused)] // XXX:temporarily unused fields
+pub struct ZomeCallParams {
+    dna_hash: String,
+    coordinator_identifier: String,
+    zome_name: String,
+    function_name: String,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct PayloadQuery {
+    pub payload: Option<String>,
+}
+
+#[tracing::instrument]
+pub async fn zome_call(
+    Path(params): Path<ZomeCallParams>,
+    Query(query): Query<PayloadQuery>,
```

Comment:
Axum handles the deserialization of the query params and query strings under the hood, handling URL encoded strings as well.

Reply:
Actually same for the payload. It's not clear from the ticket whether checking that it decodes to JSON should be a requirement for this issue but it's not on a later ticket and it's a validation check. Should we be doing that now?

Reply:
We should definitely be checking it can decode from base64

Reply:
All my comments on this thread are resolved, I'll leave it open because I didn't start it